src/app/digital/utils/ExpressionParser/TreeToCircuit.ts

```diff
@@ -59,6 +59,8 @@ export const NegatedTypeToGate: Record<InputTreeBinOpType, string> = {
     } else if (node.kind === "binop") {
         newGate.setInputPortCount(node.children.length);
         node.children.forEach(child => {
+            if (!child)
+                return;
```

Comment:
An undefined child sounds like it could be bad, probably want to throw an error here just in case cause otherwise could cause some nasty and hard to find bugs